man/tpm2_policypcr.1.md

```diff
@@ -0,0 +1,93 @@
+% tpm2_policypcr(1) tpm2-tools | General Commands Manual
+%
+% JANUARY 2018
+
+# NAME
+
+**tpm2_policypcr**(1) - Perform a policyPCR event with the TPM.
+
+# SYNOPSIS
+
+**tpm2_policypcr** [*OPTIONS*]
+
+# DESCRIPTION
+
+**tpm2_policypcr**(1) generates a policy PCR event with the TPM. It is similair
+to **tpm2_createpolicy**(1), however, it expects a session to be already
+established via **tpm2_startauthsession**(1).
+
+# OPTIONS
+
+  * **-f**, **--policy-file**=_POLICY\_FILE_:
+    File to save the policy digest.
+
+  * **-F**, **--pcr-input-file**=_PCR\_FILE_:
+    Optional Path or Name of the file containing expected pcr values for the
+    specified index. Default is to read the current PCRs per the set list.
+
+  * **-L**, **--set-list**=_PCR\_LIST_:
+    The list of pcr banks and selected PCRs' ids (0~23) for each bank.
+
+  * **-S**, **--session**=_SESSION_FILE_:
+
+    The policy session file generated via the **-S** option to
+    **tpm2_startauthsession**(1).
+
+[common options](common/options.md)
+
+[common tcti options](common/tcti.md)
+
+[supported hash algorithms](common/hash.md)
+
+[algorithm specifiers](common/alg.md)
+
+# EXAMPLES
+
+Start a *trial* session and build a PCR policy and use that policy in an object.
+Then use a *policy* session to unseal some data.
+```
+# Step 1: Create a trial session and build a pcr policy via a policyPCR event to generate
+#   a policy hash.
+#
+# Step 2: Create an object and use the policy hash as the policy to satisfy for usage.
+#
+# Step 3: Create an actual policy session and using a policyPCR event, update the session
+#  policy hash.
+#
+# Step 4: Using the actual policy session from step 3 in tpm2_unseal to unseal the object.
+#
+
+tpm2_createprimary -H e -g sha256 -G ecc -C primary.ctx
+
+tpm2_pcrlist -Q -L "sha1:0,1,2,3 -o pcr.dat
+
+handle=`tpm2_startauthsession -S session.dat | cut -d' ' -f 2-2`
```

Comment:
This is an example why I would prefer to just print the session instead of having it inside a yaml output, the pipe and cut won't be needed in that case.

Reply:
tpmNonce output will need to be added at some point, so eventually it will need multiple outputs. Id like to get some things to stdout versus always dumping to a file.

With that said tpm2_flushcontext could take -S so no need for handle to be on stdout. But I think it would be informative for users to be able to see the outputs from a command on both stdout (human readable/parseable) and via binary files (if needed as an option).